pyroomacoustics/transform/stft.py

```diff
@@ -81,6 +86,16 @@ def __init__(self, N, hop=None, analysis_window=None,
         else:
             self.hop = self.num_samples
 
+        if precision == np.float32 or precision == np.complex64 or precision == 'single':
+            self.time_dtype = np.float32
+            self.freq_dtype = np.complex64
+        elif precision == np.float64 or precision == np.complex128 or precision == 'double':
+            self.time_dtype = np.float64
+            self.freq_dtype = np.complex128
+        else:
```

Comment:
Here I think we should handle a few more cases. For example what if we call `analysis`, which detects the type automatically, on a `np.int16` array ? In the current situation, the call will fail. Ideally, we would like a cast to happen. Without more information, a cast to `np.float64` is desirable.

Most likely, the only time we will use this option is to set it to `single` when needed. The rest of the time, we'd like it set to `double`. So I think it is safe to do
```
if precision == np.float32 or precision == np.complex64 or precision == 'single':
    self.time_dtype = np.float32
    self.freq_dtype = np.complex64
else:
    self.time_dtype = np.float64
    self.freq_dtype = np.complex128
 ```
in other words, if single precision is not requested expressly, we set double precision regardless of the value of the parameter.

What do you think ?

Reply:
I agree with this point. Then the docstring would look something like this?
```
    precision : string, np.float32, np.float64, np.complex64, np.complex128, optional
        How many precision bits to use for the input.
        If 'single'/np.float32/np.complex64, 32 bits for real inputs or 64 for complex spectrum.
        Otherwise, cast to 64 bits for real inputs or 128 for complex spectrum (default)
```

Reply:
Looks good!